src/Client/ConcordiumClient.cs

```diff
@@ -0,0 +1,172 @@
+    /// </summary>
+    /// <param name="accountAddress">An address of the account to get the next sequence number of.</param>
+    /// <returns>
+    /// A tuple containing the best guess of the next sequence number for the supplied account and a boolean
+    /// indicating whether all transactions associated with the account are finalized. If the latter is the
+    /// case, then the sequence number is reliable.
+    /// </returns>
+    /// <exception cref="RpcException">The call failed.</exception>
+    public (Types.AccountSequenceNumber, bool) GetNextAccountSequenceNumber(
+        Types.AccountAddress accountAddress
+    )
+    {
+        var next = this.Raw.GetNextAccountSequenceNumber(accountAddress.ToProto());
+
+        // Return the sequence number as a "native" SDK type.
+        return (Types.AccountSequenceNumber.From(next.SequenceNumber.Value), next.AllFinal);
+    }
+
+    /// <summary>
+    /// Spawn a task which queries the chain for the next sequence number of the
+    /// account with the supplied address.
+    ///
+    /// Specifically, this is the best guess as to what the next account sequence
+    /// number (nonce) should be for the given account.
+    ///
+    /// If all account transactions are finalized then this information is reliable.
+    ///
+    /// Otherwise this is the best guess, assuming all other transactions will be
+    /// committed to blocks and eventually finalized.
+    /// </summary>
+    /// <param name="accountAddress">An address of the account to get the next sequence number of.</param>
+    /// <returns>
+    /// A task which returns a tuple containing the best guess of the next sequence number for the supplied
+    /// account and a boolean indicating whether all transactions associated with the account are finalized.
+    /// If the latter is the case, then the sequence number is reliable.
+    /// </returns>
+    /// <exception cref="RpcException">The call failed.</exception>
+    public Task<(Types.AccountSequenceNumber, bool)> GetNextAccountSequenceNumberAsync(
+        Types.AccountAddress accountAddress
+    ) =>
+        this.Raw
+            .GetNextAccountSequenceNumberAsync(accountAddress.ToProto())
+            // Continuation returning the "native" SDK type.
+            .ContinueWith(
+                next =>
+                    (
+                        Types.AccountSequenceNumber.From(next.Result.SequenceNumber.Value),
+                        next.Result.AllFinal
+                    )
+            );
+
+    #region IDisposable Support
+
+    private bool _disposedValue;
+
+    protected virtual void Dispose(bool disposing)
+    {
+        if (!this._disposedValue)
+        {
+            if (disposing)
+            {
+                this.Raw.Dispose();
+            }
+
+            this._disposedValue = true;
+        }
+    }
+
+    public void Dispose()
+    {
+        this.Dispose(true);
+        GC.SuppressFinalize(this);
+    }
+
+    ~ConcordiumClient()
```

Comment:
You should not use finalizers in .NET - whenever you use it, it will at least survive one garbage collections and be promoted to Generation 1.

Make the class `sealed` and make the disposable pattern more simple.